hilti/src/rt/types/stream.cc

```diff
@@ -417,40 +453,59 @@ std::string hilti::rt::detail::adl::to_string(const stream::SafeConstIterator& x
 void SafeConstIterator::debugPrint(std::ostream& out) const {
     int chunk = 0;
 
-    auto c = _content.lock()->head.get();
+    auto c = _chain->head();
     while ( c ) {
-        if ( c == _chunk.lock().get() )
+        if ( c == _chunk )
             break;
 
         chunk++;
-        c = c->next().get();
+        c = c->next();
     }
 
     if ( ! c )
         // Can happen if trimmed off.
         chunk = -1;
 
-    out << fmt("iterator %p: parent=%p chunk=#%d offset=%llu is_end=%d\n", this, _content.lock().get(), chunk, _offset,
+    out << fmt("iterator %p: chain=%p chunk=#%d/%p offset=%llu is_end=%d\n", this, _chain.get(), chunk, c, _offset,
                static_cast<int>(isEnd()));
 }
 
+std::string hilti::rt::detail::adl::to_string(const UnsafeConstIterator& x, adl::tag /*unused*/) {
```

Comment:
Can you add a small test for this covering the different scenarios (short & long `View`, expired & unset iterator)?

Reply:
done